dominion/src/main/java/org/codice/dominion/interpolate/Interpolator.java

```diff
@@ -0,0 +1,436 @@
+
+                  if (o instanceof String[]) {
+                    s = Arrays.toString((String[]) o);
+                  } else { // must be a string
+                    s = (String) o;
+                  }
+                } catch (IllegalAccessException | InvocationTargetException e) {
+                  s = "?";
+                }
+                return method.getName() + '=' + s;
+              }
+            });
+  }
+
+  /**
+   * Interpolates all static fields of type <code>String</code> or <code>String[]</code> that are
+   * annotated with {@link Interpolate}.
+   *
+   * @param <C> the class to interpolate
+   * @param clazz the class to interpolate
+   * @return <code>clazz</code> for chaining
+   * @throws InterpolationException if unable to interpolate the class
+   * @throws ContainerNotStagedException if an <code>"{karaf.XXXX}"</code> expression is found and
+   *     PaxExam has not yet staged the container
+   */
+  public <C> Class<C> interpolate(Class<C> clazz) {
+    ReflectionUtils.getAllAnnotationsForFieldsAnnotatedWith(clazz, Interpolate.class, true)
+        .keySet()
+        .forEach(this::interpolate);
+    return clazz;
+  }
+
+  /**
+   * Interpolates a given static field as long as it is of type <code>String</code> or <code>
+   * String[]</code>.
+   *
+   * @param field the static field to interpolate
+   * @return <code>field</code> for chaining
+   * @throws InterpolationException if unable to interpolate the field
+   * @throws ContainerNotStagedException if an <code>"{karaf.XXXX}"</code> expression is found and
+   *     PaxExam has not yet staged the container
+   */
+  public Field interpolate(Field field) {
+    final int modifiers = field.getModifiers();
+
+    if (!Modifier.isStatic(modifiers)) {
+      throw new InterpolationException("Can only interpolate annotated static fields");
+    }
+    try {
+      final Class<?> type = field.getType();
+
+      field.setAccessible(true);
+      if (String.class.equals(type)) {
+        field.set(null, interpolate((String) field.get(null)));
+      } else if (String[].class.equals(type)) {
+        field.set(null, interpolate((String[]) field.get(null)));
+      } else {
+        throw new InterpolationException(
+            "Can only interpolate annotated static fields of type String or String[]");
+      }
+    } catch (SecurityException | IllegalAccessException e) {
+      throw new InterpolationException("Unable to interpolate field: " + field, e);
+    }
+    return field;
+  }
+
+  /**
+   * Interpolates an array of strings.
+   *
+   * @param s the array of strings to interpolate
+   * @return a cloned version of the array where all elements where interpolated or <code>null
+   *     </code> if <code>s</code> is <code>null</code>
+   * @throws InterpolationException if unable to interpolate a string in the array
+   * @throws ContainerNotStagedException if an <code>"{karaf.XXXX}"</code> expression is found and
```

Comment:
`ContainerNotStagedException` is a subclass of `InterpolationException`. It can be thrown out as a more specific exception when an interpolation of specific expressions occurs and those expressions requires the container to be staged. `{karaf.home}` is such a case when using the Dominion PaxExam driver.